lib/Sema/ConstraintSystem.h

```diff
@@ -407,8 +407,10 @@ enum ScoreKind {
   SK_ScalarPointerConversion,
   /// A conversion from an array to a pointer of matching element type.
   SK_ArrayPointerConversion,
+  /// A conversion to an empty existential type ('Any' or '{}').
```

Comment:
Yes. I'll definitely be revisiting this area at some point in the nearish future.